services/plugins/npmjs-plugins-source.ts

```diff
@@ -0,0 +1,113 @@
+import * as cheerio from "cheerio";
+import {PluginsSourceBase} from "./plugins-source-base";
+
+export class NpmjsPluginsSource extends PluginsSourceBase implements IPluginsSource {
+	private static NPMJS_ADDRESS = "http://npmjs.org";
+
+	private _hasReturnedTheInitialPlugins: boolean;
+	private _keywords: string[];
+
+	constructor(private $httpClient: Server.IHttpClient,
+		private $childProcess: IChildProcess,
+		private $hostInfo: IHostInfo,
+		private $progressIndicator: IProgressIndicator,
+		private $logger: ILogger,
+		private $errors: IErrors) {
+		super();
+	}
+
+	public initialize(projectDir: string, keywords: string[]): IFuture<void> {
+		return (() => {
+			super.initialize(projectDir, keywords).wait();
+
+			this._keywords = keywords;
+
+			this._plugins = this.getPluginsFromNpmjs(keywords, 0).wait();
+		}).future<void>()();
+	}
+
+	public getPlugins(page: number, count: number): IFuture<IBasicPluginInformation[]> {
+		return ((): IBasicPluginInformation[] => {
+			if (!this._hasReturnedTheInitialPlugins && page === 0) {
```

Comment:
I don't quite understand this logic - `this._hasReturnedTheInitialPlugins`'s value is never changed so it'll always be `undefined`. Moreover, you return `this._plugins` here and this object may contain multiple pages worth of information because of the line `this._plugins = this._plugins.concat(result);` below